File: 3880.minimum-absolute-difference-between-two-values/solution.go
func minAbsoluteDifference(nums []int) int {
	minDiff := -1
	last1, last2 := -1, -1

	for i, num := range nums {
		if num == 1 {
			last1 = i
			if last2 != -1 {
				diff := i - last2
				if minDiff == -1 || diff < minDiff {
					minDiff = diff
				}
			}
		} else if num == 2 {
			last2 = i
			if last1 != -1 {
				diff := i - last1
				if minDiff == -1 || diff < minDiff {
					minDiff = diff
				}
			}
		}
	}

	return minDiff    
}

File: 3880.minimum-absolute-difference-between-two-values/solution.kt
class Solution {
    fun minAbsoluteDifference(nums: IntArray): Int {
        var i1: Int? = null
        var i2: Int? = null
        var minDiff = Int.MAX_VALUE

        fun minimize(i: Int?, j: Int) {
            if (i != null) {
                minDiff = minOf(j - i, minDiff)
            }
        }

        nums.forEachIndexed { i, num ->
            if (num == 1) {
                minimize(i2, i)
                i1 = i
            } else if (num == 2) {
                minimize(i1, i)
                i2 = i
            }        
        }

        return if (minDiff == Int.MAX_VALUE) -1 else minDiff
    }
}